src/rpc-client.ts

```diff
@@ -1,95 +1,263 @@
-const http = require('http');
-const { BufferUtils } = require('@nimiq/core');
+import { NimiqRPCClient, Transaction } from '@blouflash/nimiq-rpc';
+import { KeyPair, Address, TransactionBuilder, BufferUtils } from '@nimiq/core';
 
-class RpcClient {
-    constructor(host, port) {
+/** Parameters for sending a transaction */
+interface BaseTransactionParams {
+    wallet: string;
+    recipient: string;
+    value: number;
+    fee: number;
+    data?: string;
+}
+
+type TransactionParams = BaseTransactionParams;
+
+export interface RawTransactionParams {
+    sender: KeyPair;
+    recipient: string;
+    value: number;
+    fee?: number;
+    data?: string;
+}
+
+/**
+ * RPC Client for interacting with the Nimiq node
+ * Provides methods for blockchain queries and wallet operations
+ */
+export class RpcClient {
+    private _host: string;
+    private _port: string;
+    private _client: NimiqRPCClient;
+
+    /**
+     * Creates a new RPC client instance
+     * @param host - The hostname of the Nimiq node
+     * @param port - The port number of the Nimiq node
+     */
+    constructor(host: string, port: string) {
         this._host = host;
         this._port = port;
+        this._client = new NimiqRPCClient({
+            httpUrl: `http://${this._host}:${this._port}`,
+            wsUrl: `ws://${this._host}:${this._port}/ws`
+        });
     }
 
-    async isConnected() {
+    /**
+     * Checks if the client is connected to the node
+     * @returns Promise resolving to true if connected
+     */
+    async isConnected(): Promise<boolean> {
         try {
-            return !!await this.getBlockHeight();
+            const response = await this._client.blockchain.getBlockNumber();
+            return response !== undefined;
         } catch (e) {
             return false;
         }
     }
 
-    async getBlockHeight() {
-        return this._jsonRpcFetch('blockNumber');
+    /**
+     * Checks if consensus is established with the network
+     * @returns Promise resolving to consensus status
+     */
+    async isConsensusEstablished(): Promise<boolean> {
+        const response = await this._client.consensus.isConsensusEstablished();
+        return Boolean(response.data);
```

Comment:
Could be slightly nicer with destructuring.
```suggestion
        const { data } = await this._client.consensus.isConsensusEstablished();
        return Boolean(data);
```